### ISSUE TITLE
Website rendering not matching pkgdown rendering or html preview

### ISSUE THREAD

**Issue by samuel-marsh** [r-lib/pkgdown#2324]
Hi pkgdown team,

I'm not entirely sure this is best discussed here so if there is better forum for this issue please let me know!  But since pkgdown is source of website I thought best to start here.

I have pkgdown site for my R package in which some of the articles rely of pre-rendered images.  When I use `build_site` locally the preview renders just fine and images are present.  Additionally if I copy link from repo and use https://htmlpreview.github.io/ it also renders the images properly (using: https://github.com/samuel-marsh/scCustomize/blob/master/docs/articles/Read_and_Write_Functions.html).
![image](https://github.com/r-lib/pkgdown/assets/38284410/5d4d0741-701d-4c25-9a8f-4e4820eeb27f)

However, visiting the deployed site the images are absent.
https://samuel-marsh.github.io/scCustomize/articles/Read_and_Write_Functions.html
![image](https://github.com/r-lib/pkgdown/assets/38284410/0dde295e-0de4-4593-9da6-baa7619767b0)

I'm confused as to the issue that could be causing such a discrepancy and any help would be greatly appreciated!

Best,
Sam

**Comment by maelle** [comment]
I see there's a relative path in https://github.com/samuel-marsh/scCustomize/blob/ff1cf1ba104f97e4e1037883f2fefd0df1f2534b/vignettes/articles/Read_and_Write_Functions.Rmd#L93. 

On the local website, the root of the website is the root of the pkgdown website whereas on GitHub pages, the root of https://samuel-marsh.github.io/scCustomize/articles/Read_and_Write_Functions.html is https://samuel-marsh.github.io/, that's the difference explaining the discrepancy.

Here's an example, in another package, that works: https://github.com/allisonhorst/palmerpenguins/blob/c19a904462482430170bfe2c718775ddb7dbb885/vignettes/pca.Rmd#L287

Hope this helps

**Comment by samuel-marsh** [comment]
Hi @maelle 

Thanks for quick reply!

So I gave it try using URL to the png file (in master branch of repo) to `knitr::include_graphics` and it successfully knitted that but then failed to render the png in output.

I'm little confused as to how I can modify the relative paths given that `pkgdown::build` fails locally if paths are not pointed in the right place.

Best,
Sam

**Comment by samuel-marsh** [comment]
Adding some previous discussions we've had on related issues both here and on RStudio forum.

https://github.com/r-lib/pkgdown/issues/2246#issuecomment-1334453525
https://community.rstudio.com/t/r-cmd-check-failing-vignette-erroring/123830/4

I should mention that these Rmd are being built as articles as they rely on loading datasets that are too big to be included on GitHub easily.  This was originally issue with vignettes building and therefore changed them to articles per your suggestion in RStudio post.  They are built and then pushed to GitHub.  Hence why I was surprised that the website itself wasn't rendering the images when using html preview of the .html file seemed to work just fine.

Finally I seemed to have lucked out and wayback machine actually contains an archive of this specific page and it shows things rendering fine in Nov 2021 but next available archive Dec 2022 they are not working anymore.  I haven't changed the file paths or file images in this vignette since I got things working when I released package.  So maybe some update on pkgdown or GitHub pages end?  Just trying to think what it could be.
Nov 2021 (Working):
https://web.archive.org/web/20211117193849/https://samuel-marsh.github.io/scCustomize/articles/Read_and_Write_Functions.html

Dec 2022 (Not working):
https://web.archive.org/web/20221231144008/https://samuel-marsh.github.io/scCustomize/articles/Read_and_Write_Functions.html

Best,
Sam

**Comment by maelle** [comment]
Does using the same solution as https://github.com/allisonhorst/palmerpenguins/blob/c19a904462482430170bfe2c718775ddb7dbb885/vignettes/pca.Rmd#L287 work? (sorry I don't have more time to explore)

**Comment by samuel-marsh** [comment]
Hi @maelle,

No worries, and thanks for all the help!!

so I thought I had tried using `<img src=../etc` but maybe I had typed it wrong.

Regardless of directory structure and moving things around using `knitr::include_graphics` always failed to render when pushed to GitHub.

However, switching to the direct sourcing of image outside of R code chunk seems to have done the trick.

Thanks again!!
Sam
